### ISSUE TITLE
Clojure REPL throwing error, even the same function works in "lein repl"

### ISSUE THREAD

**Issue by JohnPostlethwait** [LightTable/LightTable#1829]
This function call comes back with an error: https://github.com/swannodette/lt-cljs-tutorial/blob/master/lt-cljs-tutorial.cljs#L619

```
TypeError: Cannot call method 'call' of undefined
    at eval (/Users/johnpostlethwait/Documents/workspace/lt-cljs-tutorial/lt-cljs-tutorial.cljs[eval6]:625:27)
    at eval (native)
    at Function.eval (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:35385:461)
    at b (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:6188:14)
    at a (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:6234:18)
    at cljs.core.do_dispatch (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:13583:26)
    at cljs.core.MultiFn.cljs$core$IMultiFn$_dispatch$arity$2 (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:13655:32)
    at cljs.core._dispatch (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:13568:14)
    at a (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:13662:32)
    at b [as call] (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:13666:14)
```
![screen shot 2015-01-23 at 12 56 37 am](https://cloud.githubusercontent.com/assets/39127/5872000/b9cd9060-a29a-11e4-9d83-87eeb36e2a8e.png)

Even though this same function is working in `lein repl`:

```
user=> (defn defaults
  #_=>   ([x] (defaults x :default))
  #_=>   ([x y] [x y]))
#'user/defaults
user=> (defaults :explicit)
[:explicit :default]
```

**Comment by cldwalker** [comment]
Hi. It seems you haven't evaled `defaults` yet. After you eval the defn, your (defaults) evals should work

**Comment by JohnPostlethwait** [comment]
That was it.

Perhaps this should remain open to have a clearer error? Looking at the trace, it was all through LightTable javascript files, and isn't very clear that the issue was on my end.

For instance, when I run code without evaluating the namespace first, I get a reasonable message:

```
ReferenceError: lt_cljs_tutorial is not defined
    at eval (/Users/johnpostlethwait/Documents/workspace/lt-cljs-tutorial/lt-cljs-tutorial.cljs[eval1]:625:1)
    at eval (native)
    at Function.eval (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:35385:461)
    at b (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:6188:14)
    at a (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:6234:18)
    at cljs.core.do_dispatch (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:13583:26)
    at cljs.core.MultiFn.cljs$core$IMultiFn$_dispatch$arity$2 (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:13655:32)
    at cljs.core._dispatch (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:13568:14)
    at a (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:13662:32)
    at b [as call] (file:///Applications/LightTable.app/Contents/Resources/app.nw/core/node_modules/lighttable/bootstrap.js:13666:14)
```

**Comment by cldwalker** [comment]
I agree that ClojureScript errors can be hard to parse. However that should be solved with the language, not by capturing and rethrowing them for a one-off case. If you'd like to start a wiki page that explains common ClojureScript errors that would be helpful for newcomers to the language.

**Comment by JohnPostlethwait** [comment]
@cldwalker Gotcha, thanks. I wasn't sure if this was something in the core ClojureScript stack, or something being recast by LightTable. (If the "ReferenceError : lt_cljs_tutorial" was any indication, I am new to this.) :tongue: 

Thanks for all the help!
